Simumatik/simumatik-gateway
# Simulink driver

This is the first draft for the Simulink driver.
These are the steps to test it:
1. Install MATLAB along with Simulink (hopefully no additional Add-Ons are required)
2. Start MATLAB as administrator and run the two following commands in the main Command window:
```
cd (fullfile(matlabroot,'extern','engines','python'))
system('python setup.py install')
```
3. Now restart Matlab as usual (no admin)
4. Then run `matlab.engine.shareEngine` in the same Command window
5. Open the file in this archive: [simulink_test1.zip](https://github.com/Simumatik/simumatik-gateway/files/6863791/simulink_test1.zip)
6. In the new Simulink window, press Run and wait for it to compile.
7. Now you should be able to run `python src/tests/test_simulink.py`. Simulink will automatically run and stop after testing.

There is also a system in Simumatik called 'Simulink test' with a component I have used for testing during development.
